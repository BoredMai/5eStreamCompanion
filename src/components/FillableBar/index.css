:local(.fillable-bar) {
  align-items: center;
  border: 2px solid black;
  display: flex;
  justify-content: center;
  height: 1.5rem;
  margin-left: 3rem;
  position: relative;
  width: 15rem;
  z-index: 2;
}

:local(.bar-fill) {
  height: 1.5rem;
  left: 0;
  top: 0;
  position: absolute;
  z-index: -1;
}

:local(.bar-label) {
  left: -3rem;
  position: absolute;
}

:global(.battle-mode) :local(.fillable-bar) {
  width: 19rem;
}
