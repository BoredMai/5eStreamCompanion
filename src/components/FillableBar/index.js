import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { barFill, barLabel, fillableBar } from './index.css';

export default class FillableBar extends Component {
  render() {
    const { className, color, current, label, max } = this.props;
    const width = Math.max(0, Math.min((current * 100) / max, 100)) + '%';
    const style = {
      backgroundColor: color,
      width,
    };
    return (
      <div className={className}>
        <div className={fillableBar}>
          <span className={barLabel}>{label}</span>
          <div className={barFill} style={style} />
          <span>
            {current} / {max}
          </span>
        </div>
      </div>
    );
  }

  static propTypes = {
    className: PropTypes.string,
    color: PropTypes.string.isRequired,
    current: PropTypes.number.isRequired,
    label: PropTypes.string,
    max: PropTypes.number.isRequired,
  };
}
