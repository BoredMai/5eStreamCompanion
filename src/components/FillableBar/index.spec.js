import React from 'react';
import { shallow } from 'enzyme';
import FillableBar from '.';

describe('<FillableBar />', () => {
  let wrapper;
  let props = {
    color: '#ff6666',
    current: 100,
    max: 100,
  };

  beforeEach(() => {
    wrapper = shallow(<FillableBar {...props} />);
  });

  it('renders properly', () => {
    expect(wrapper.getElement()).toMatchSnapshot();
  });

  it('shrinks bar fill when current < max', () => {
    wrapper.setProps({ current: 40 });
    const barFill = wrapper.find('.barFill');
    const {
      style: { width },
    } = barFill.props();
    expect(width).toBe('40%');
  });

  it('does not overflow bar fill when current > max', () => {
    wrapper.setProps({ current: 120 });
    const barFill = wrapper.find('.barFill');
    const {
      style: { width },
    } = barFill.props();
    expect(width).toBe('100%');
  });

  it('does not underflow bar fill when current < 0', () => {
    wrapper.setProps({ current: -20 });
    const barFill = wrapper.find('.barFill');
    const {
      style: { width },
    } = barFill.props();
    expect(width).toBe('0%');
  });
});
