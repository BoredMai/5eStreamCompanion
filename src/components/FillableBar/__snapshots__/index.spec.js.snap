// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FillableBar /> renders properly 1`] = `
<div>
  <div
    className="fillableBar"
  >
    <span
      className="barLabel"
    />
    <div
      className="barFill"
      style={
        Object {
          "backgroundColor": "#ff6666",
          "width": "100%",
        }
      }
    />
    <span>
      100
       / 
      100
    </span>
  </div>
</div>
`;
